src/components/Pagination/index.jsx

```diff
@@ -21,19 +22,28 @@ const Pagination = ({
     siblingCount,
     pageSize,
   });
+  const { handlePaginationQueryParamsChange } = usePaginationQueryParams();
+
+  if (!navigate) navigate = handlePaginationQueryParamsChange;
 
   if (pageNo === 0 || paginationRange.length < 2) {
     return null;
   }
 
   const onNext = () => {
-    !isLastPage && navigate(pageNo + 1);
+    if (isLastPage) return;
+
+    navigate(pageNo + 1);
   };
 
   const onPrevious = () => {
-    !isFirstPage && navigate(pageNo - 1);
+    if (isFirstPage) return;
+
+    navigate(pageNo - 1);
   };
 
+  const onPaginationItemClick = page => navigate(page);
```

Comment:
Why do you need such a proxy function?
Why not directly use `navigate` wherever required?